### PR TITLE
Hw 1 jingyi han

### PR DESCRIPTION
**Pull request by jingyihan**

### CODE REVIEW

File: Homeworks/HW1/JINGYI_HAN/game_of_life.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+def evolve(initial_state):
+    # fill this out
+    neighbours = [[0 for j in range(len(initial_state[0]))] for i in range(len(initial_state))]
+    for i in range(len(initial_state)):
+        for j in range(len(initial_state[0])):
+            if initial_state[i][j] == 1:
```
Comment (pusukuri) [inline Homeworks/HW1/JINGYI_HAN/game_of_life.py:6]: Please provide comments that explain the code a bit..

File: Homeworks/HW1/JINGYI_HAN/game_of_life.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                for k in [[0,-1],[0,1],[-1,-1],[1,1],[-1,0],[1,0],[-1,1],[1,-1]]:
+                    if i+k[0] >= 0 and i+k[0] < len(initial_state) and j+k[1] >= 0 and j+k[1] < len(initial_state[0]):
+                        neighbours[i+k[0]][j+k[1]] += 1
+
+    for i in range(len(initial_state)):
+        for j in range(len(initial_state[0])):
+            if initial_state[i][j] == 1 and (neighbours[i][j] != 2 and neighbours[i][j] != 3):
```
Comment (pusukuri) [inline Homeworks/HW1/JINGYI_HAN/game_of_life.py:13]: It would be good to provide comments that explain the meaning of "==1" etc..

**Review by pusukuri — APPROVED** [review]
Overall good. Great job.

File: Homeworks/HW1/JINGYI_HAN/game_of_life.py (hunk 1 above)
Comment (jingyihan) [reply Homeworks/HW1/JINGYI_HAN/game_of_life.py:6]: Thanks, Kishore! Will include comments in future submissions!

**Pull request merged**